resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesConf.scala

```diff
@@ -148,7 +149,8 @@ private[spark] class KubernetesExecutorConf(
     executorCustomLabels ++ presetLabels
   }
 
-  override def environment: Map[String, String] = sparkConf.getExecutorEnv.toMap
+  override def environment: Map[String, String] = sparkConf.getExecutorEnv.filter(
```

Comment:
This is a nit, but you're using a filter in order to apply a check to each element. A foreach would be more appropriate and avoid the need to "filter" when the predicate won't matter.

Reply:
My goal is to filter and remove irregular elements in executorEnv, and to retain qualified elements;

 I think 'filter' is more appropriate. 

Isn't it more complicated to use "foreach"?

Reply:
Oops, I misread this. Disregard.